src/beamlime/applications/handlers.py

```diff
@@ -157,8 +159,11 @@ def from_args(
 class DataReductionHandler(HandlerInterface):
     """Data reduction handler to process the raw data."""
 
-    def __init__(self, workflow: StatelessWorkflow) -> None:
+    def __init__(
+        self, workflow: StatelessWorkflow, reuslt_registry: ResultRegistry | None = None
+    ) -> None:
         self.workflow = workflow
+        self.result_registry = {} if reuslt_registry is None else reuslt_registry
```

Comment:
Do we need to worry about excessive memory usage if we keep this registry around?

Reply:
Not really, they'll be usually histograms, so a few hundreds of numbers.